src/ingestion/ec2_describe_reserved_instance_modifications.py

```diff
@@ -0,0 +1,154 @@
+    if CACHED_AWS_CLIENT is None:
+        CACHED_AWS_CLIENT = get_cached_aws_client()
+    target_role = f'arn:aws:iam::{account[0]}:role/{AWS_AUDIT_ROLE_NAME}'
+    dest_role = CACHED_AWS_CLIENT.assume_role(
+        RoleArn=target_role,
+        RoleSessionName=os.environ['EC2_INSTANCE_LIST_AWS_AUDIT_DESTINATION_ROLE_SESSION_NAME'],
+        ExternalId=os.environ['AWS_AUDIT_DESTINATION_ROLE_ARN_EXTERNALID']
+    )
+    ec2_session = boto3.Session(
+        aws_access_key_id=dest_role['Credentials']['AccessKeyId'],
+        aws_secret_access_key=dest_role['Credentials']['SecretAccessKey'],
+        aws_session_token=dest_role['Credentials']['SessionToken']
+    )
+    return ec2_session
+
+
+GET_ACCOUNTS_QUERY = f"""
+SELECT account:Id::string, account:Name
+FROM {AWS_ACCOUNTS_TABLE}
+WHERE timestamp = (
+  SELECT MAX(timestamp) FROM {AWS_ACCOUNTS_TABLE}
+);
+"""
+
+
+def get_accounts_list(sf_client):
+    res = sf_client.cursor().execute(GET_ACCOUNTS_QUERY).fetchall()
+    return res
+
+
+LOAD_INSTANCE_LIST_QUERY = f"""
+INSERT INTO {INSTANCES_TABLE} (timestamp, instance)
+SELECT '{{snapshotclock}}'::timestamp_ltz, PARSE_JSON(column1)
+FROM VALUES {{format_string}}
+"""
+
+
+def get_data_worker(account):
+    try:
+        ec2_session = get_aws_client(account)
+        instances = []
+        try:
+            ec2_regions = [region['RegionName'] for region in ec2_session.client('ec2').describe_regions()['Regions']]
+        except Exception as e:
+            log.error(f"ec2_describe_reserved_instance_modifications: account: {account[1]} exception: {e}")
+            return None
+        for region in ec2_regions:
+            try:
+                client = ec2_session.client('ec2', region_name=region)
+                paginator = client.get_paginator('describe_reserved_instances_modifications')
+                page_iterator = paginator.paginate()
+                region = [instance for page in page_iterator for instance in page['ReservedInstancesModifications']]
+                instances.extend(region)
+            except Exception as e:
+                log.error(f"ec2_describe_reserved_instance_modifications: account: {account[1]} exception: {e}")
+                return None
+        instance_list = [json.dumps({**instance,"AccountId":account[0]}, default=str) for instance in instances]
+        log.info(f"ec2_describe_reserved_instance_modifications: account: {account[1]} instances: {len(instance_list)}")
+        return instance_list
+    except Exception as e:
+        log.error(f"ec2_describe_reserved_instance_modifications: account: {account[1]} exception: {e}")
+        return None
+
+def get_data(accounts_list):
+    start = datetime.datetime.now()
+    instance_list_list = Pool(4).map(get_data_worker, accounts_list)
+    instance_list = [x for l in instance_list_list if l for x in l]
+    if instance_list:
+        sf_client = get_snowflake_client()
+        instance_groups = groups_of(15000,instance_list)
```

Comment:
we're repeating this in a bunch of places -- it would be better to incorporate it into `db.insert` helper